docs/pages/technical-specs/expo-updates-0.md

```diff
@@ -120,9 +136,10 @@ type Asset = {
     * `hash`: SHA-256 hash of the file to guarantee integrity.
     * `key`: Key used to reference this asset from the update's application code. This key, for example, may be generated by a separate build step that processes the application code, such as a bundler.
     * `contentType`: The MIME type of the file as defined by [RFC 2045](https://tools.ietf.org/html/rfc2045). e.g. `application/javascript`, `image/jpeg`.
-    * `fileExtension`: The suggested extension to use when a file is saved on a client. Some platforms, such as iOS, require certain file types to be saved with an extension. The extension MUST be prefixed with a `.`. e.g. `.jpeg`. In some cases, such as the launchAsset, this field will be ignored in favor of a locally determined extension. If the field is omitted and there is no locally stipulated extension, the asset will be saved without an extension, e.g. `./filename` with no `.` at the end. 
+    * `fileExtension`: The suggested extension to use when a file is saved on a client. Some platforms, such as iOS, require certain file types to be saved with an extension. The extension MUST be prefixed with a `.`. e.g. `.jpeg`. In some cases, such as the launchAsset, this field will be ignored in favor of a locally determined extension. If the field is omitted and there is no locally stipulated extension, the asset will be saved without an extension, e.g. `./filename` with no `.` at the end.
     A conforming client SHOULD prefix a file extension with a `.` if a file extension is not empty and missing the `.` prefix.
     * `url`: Location at which the file may be fetched.
+    * `signature`: MAY contain the signature of the asset to be used during the validation step of [code signing](#code-signing). Alternatively, the signature may be specified in an `expo-signature` response header on the asset request itself. The behavior of specifying both is undefined, but one or the other must be specified if the request for the manifest contained the `expo-expects-signature` header.
```

Comment:
If the client sets `expo-expects-signature` when requesting a manifest but chooses not to set it when requesting an asset (that is, the client cares about manifest signing but not code signing since they trust HTTPS), it seems acceptable IMO for the asset's response not to include an `expo-signature` header.

Reply:
What is the benefit of an `asset` signature given that we can verify it by comparing the (sha256)`hash`s?

Reply:
> What is the benefit of an `asset` signature given that we can verify it by comparing the (sha256)`hash`s?

Great question. I don't have an answer, and since the sha256s are included in the manifest which is signed, we're essentially doing a rsa-sha256 of the assets just by signing the manifest. Maybe assets other than the manifest don't need code signing?

Reply:
Now that I think about this, we definitely don't need to sign the assets independently, and actually not doing so cleans up this weird header-or-manifest-specification of asset signatures. I'll remove it from the spec. We can definitely rely upon the hashes.